src/main/java/com/epam/grid/engine/provider/utils/slurm/job/SacctCommandParser.java

```diff
@@ -0,0 +1,126 @@
+/*
+ *
+ *  * Copyright 2022 EPAM Systems, Inc. (https://www.epam.com/)
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License");
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *     http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ *
+ */
+
+package com.epam.grid.engine.provider.utils.slurm.job;
+
+import com.epam.grid.engine.entity.job.slurm.SlurmJob;
+import com.epam.grid.engine.exception.GridEngineException;
+import com.epam.grid.engine.provider.utils.slurm.DateUtils;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import org.springframework.http.HttpStatus;
+
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.epam.grid.engine.utils.TextConstants.COMMA;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class SacctCommandParser {
+
+    private static final String STANDARD_SLURM_DELIMETER = "\\|";
+    private static final String FAILED_TO_PARSE_JOB_DATA = "failed to parse job data";
+    private static final String SLURM_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss";
+
+    public static List<String> parseSlurmJob(final String jobString) {
+        final String[] jobArray = jobString.split(STANDARD_SLURM_DELIMETER);
+        if (jobArray.length != 52) {
```

Comment:
`52` seems like a magic number. I would recommend making it configurable via `application.properties`, add to `SlurmJobProvider` as a field and pass to the `SacctCommandParser.parseSlurmJob`.

Besides, this method doesn't contain that much logic - maybe it could be moved to `mapJobDataToSlurmJob` (this way `mapJobDataToSlurmJob` should accept a `String` instead of `List<String>`)

Reply:
done